src/DataProtection/DataProtection/src/KeyManagement/KeyRingBasedDataProtector.cs

```diff
@@ -142,36 +143,17 @@ public byte[] Protect(byte[] plaintext)
             }
         }
 
-        // Helper function to read a GUID from a 32-bit alignment; useful on architectures where unaligned reads
-        // can result in weird behaviors at runtime.
-        private static Guid Read32bitAlignedGuid(void* ptr)
+        private static Guid ReadGuid(void* ptr)
         {
-            Debug.Assert((long)ptr % 4 == 0);
-
-            Guid retVal;
-            if (BitConverter.IsLittleEndian)
-            {
-                ((int*)&retVal)[0] = ((int*)ptr)[0];
-                ((int*)&retVal)[1] = ((int*)ptr)[1];
-                ((int*)&retVal)[2] = ((int*)ptr)[2];
-                ((int*)&retVal)[3] = ((int*)ptr)[3];
-            }
-            else
-            {
-                // The first 8 bytes of the Guid hold one 32-bit integer
-                // and two 16-bit integers that must be byte-swapped.
-                ((byte*)&retVal)[0] = ((byte*)ptr)[3];
-                ((byte*)&retVal)[1] = ((byte*)ptr)[2];
-                ((byte*)&retVal)[2] = ((byte*)ptr)[1];
-                ((byte*)&retVal)[3] = ((byte*)ptr)[0];
-                ((byte*)&retVal)[4] = ((byte*)ptr)[5];
-                ((byte*)&retVal)[5] = ((byte*)ptr)[4];
-                ((byte*)&retVal)[6] = ((byte*)ptr)[7];
-                ((byte*)&retVal)[7] = ((byte*)ptr)[6];
-                ((int*)&retVal)[2] = ((int*)ptr)[2];
-                ((int*)&retVal)[3] = ((int*)ptr)[3];
-            }
-            return retVal;
+#if NETCOREAPP
+            // Performs appropriate endianness fixups
+            return new Guid(new ReadOnlySpan<byte>(ptr, sizeof(Guid)));
+#elif NETSTANDARD2_0 || NET461
```

Comment:
Should we be consistent with our ifdefs?  When I added some of these we just assumed things were netstandard/net461 in https://github.com/dotnet/aspnetcore/commit/5be10825f7d85ff6c6ba04939e17583c8ba4d4c7#diff-455aad1046fc7a0758d4c1e9f42260c7da7b80dfebb87240e4243d638b2ab59b

Reply:
Or at least we should use NETFRAMEWORK per: https://github.com/dotnet/aspnetcore/pull/36328#discussion_r708509338

Reply:
@HaoK yea, we should be consistent. I copied these from elsewhere in the codebase. Let me update it to `#if NETSTANDARD2_0 || NETFRAMEWORK`.